src/model/likelihood.rs

```diff
@@ -16,79 +17,70 @@ impl LatentVariableModel {
-                        .ln_add_exp(prob_sample_alt_case.ln_one_minus_exp() + observation.prob_ref);
-                assert!(!prob_case.is_nan());
-
-                // Step 4: total probability
-                let total = (observation.prob_mapping + prob_control.ln_add_exp(prob_case))
-                    .ln_add_exp(observation.prob_mapping.ln_one_minus_exp() + prob_mismapped);
-                //println!("afs {}:{}, case {:?} control {:?} total {:?}", allele_freq_case, allele_freq_control, prob_case, prob_control, total);
-                assert!(!total.is_nan());
-                total
-            }
-            (None, purity) => {
-                // no AF for control sample given
-                if let Some(purity) = purity {
-                    assert!(
-                        purity == LogProb::ln_one(),
-                        "no control allele frequency given but purity is not 1.0"
-                    );
-                }
-                // Step 1: calculate probability to sample from alt allele
-                let prob_sample_alt = LogProb(allele_freq_case.ln()) + observation.prob_sample_alt;
-
-                // Step 2: read comes from case sample and is correctly mapped
-                let prob_case = (prob_sample_alt + observation.prob_alt)
-                    .ln_add_exp(prob_sample_alt.ln_one_minus_exp() + observation.prob_ref);
-                assert!(!prob_case.is_nan());
-
-                // Step 3: total probability
-                let total = (observation.prob_mapping + prob_case)
-                    .ln_add_exp(observation.prob_mapping.ln_one_minus_exp() + prob_mismapped);
-                assert!(!total.is_nan());
-                total
-            }
-            (Some(_), None) => panic!("control allele frequency given but purity not defined"),
-        }
+        // Step 1: probability to sample observation: AF * placement induced probability
+        let prob_sample_alt_case = allele_freq_case + observation.prob_sample_alt;
+        let prob_sample_alt_control = allele_freq_control + observation.prob_sample_alt;
+
+        // Step 2: read comes from control sample and is correctly mapped
+        let prob_control = self.impurity.unwrap()
+            + (prob_sample_alt_control + observation.prob_alt)
+                .ln_add_exp(prob_sample_alt_control.ln_one_minus_exp() + observation.prob_ref);
+        assert!(!prob_control.is_nan());
+
+        // Step 3: read comes from case sample and is correctly mapped
+        let prob_case = self.purity.unwrap()
+            + (prob_sample_alt_case + observation.prob_alt)
+                .ln_add_exp(prob_sample_alt_case.ln_one_minus_exp() + observation.prob_ref);
+        assert!(!prob_case.is_nan());
+
+        // Step 4: total probability
+        let total = (observation.prob_mapping + prob_control.ln_add_exp(prob_case))
+            .ln_add_exp(observation.prob_one_minus_mapping + prob_mismapped);
```

Comment:
Let's just write
```suggestion
            .ln_add_exp(observation.prob_mismapping);
```
saving another operation.

Reply:
With the naming, I wanted to avoid a confusion with prob_mismapped. Should we then completely remove `prob_mismapped()` as it does not seem to server any purpose?

Reply:
The refactored and `cargo fmt`ed new code is up and CI testing...